syntax = "proto3";

package eleitor;

enum TipoVinculo {
    TIPO_DESCONHECIDO = 0;
    DISCENTE = 1;
    DOCENTE = 2;
}

enum Status {
    STATUS_DESCONHECIDO = 0;
    ATIVO = 1;
    INATIVO = 2;
    SUSPENSO = 3;
}


message Vinculo {
    int32 id = 1;
    string matricula = 2;
    TipoVinculo tipo = 3;
    int32 id_pessoa = 4;
    Curso curso = 5;
}

message Etapa {
    int32 etapa = 1;
    string turno = 2;
}

message Curso {
    int32 id = 1;
    string nome = 2;
    string descricao = 3;
    Etapa etapa = 4;
}

message Pessoa {
  int32 id = 1;
  string cpf = 2;
  string email = 3;
  string data_nascimento = 4;
  string nome = 5;
  repeated Vinculo vinculos = 6;
}

message Eleitor {
  Pessoa pessoa = 1;
  Status status = 2;
}

message GrupoEleitores {
    int32 id = 1;
    string nome = 2;
    string descricao = 3;
    bool ativo = 4;
    repeated Eleitor lista_eleitores = 5;
}


service GrupoEleitorService {
  rpc EnviarGrupo(GrupoEleitores) returns (EnviarGrupoResponse);
}

message EnviarGrupoResponse {
  string mensagem = 1;
}